### PR TITLE
Grafana fixes

### PR DESCRIPTION
**Pull request by simondeziel**
Connecting null values should help with instances that are not always online.

### CODE REVIEW

**Review by tomponline — APPROVED** [review]

**Comment by tomponline** [comment]
Does this address https://github.com/canonical/lxd/issues/13165 by any chance?

**Comment by simondeziel** [comment]
> Does this address #13165 by any chance?

Unfortunately not, I have yet to look into this one.

**Comment by nchorekchyan** [comment]
As an FYI: The issue, at least in my case is as follows:

TLDR: You are using the wrong variable to filter the Loki logs.

Details:

Scenario: Multi-Node LXD Cluster. 
Prometheus has one job that collects from the cluster. Loki does not report by cluster, rather by instance.

The dashboard filters by $job, which is the prometheus job name configured in prometheus's config.yml filer. 

Loki references by server's hostname (I think), which is not going to match the job name, and will have multiple instances as opposed to one job (unless your jobs are not by cluster, but by host, in which case the job name would still need to match)

The variable $job  (name: job) is used for filtering the Loki 

The expression
`"expr": "{app=\"lxd\", type=\"lifecycle\", instance=\"$job\"}",
`
Is what does the filtering for the Loki panels, which means it will never work in a clustered environment, or if the job name is different from the server in a standalone.

I was able to work around this by adding a separate variable, called $hosts, which is a simple query defined by the instance label (which in this case means the hostname of the cluster member, not to be confused with the name of the server in the cluster)

So the updated Loki queries become  (I switched to regex to allow for multiple hosts to be selected)

`          "expr": "{app=\"lxd\", type=\"lifecycle\", instance=~\"$hosts\"}",
`

The only downside to the above is I can't figure out how to get it to do "All" in a single button. I'm new to Grafana but I'm familiar with splunk which is pretty similar. When I do the "all" option in the variable, and select all, it just comes up as blank, but if I select my hosts independently, they work normally.

This of course does not apply filtering to the prometheus based queries on the dashboard, but it's late and I just figured this out and dont want to go through that exercise yet :-).

Hopefully this helps and/or is relevant.
